Core/Scripts: zone_terokkar_forest cleanup

tkrokli opened the pull request:
**Changes proposed**:
- moved spell and quest IDs into enum
- replaced hardcoded text with DB gossip
- restored correct gossip option order for Skull Pile

**Target branch(es)**: 3.3.5 & 6.x

**Issues addressed**: General cleanup of the `zone_terokkar_forest.cpp` script file

**Tests performed**: (Does it build? Tested in-game?) Yes.

**Known issues and TODO list**:
Several (if not all) of these individual scripts could likely be converted into SAI, but the reason I have not done it is because I do not know how to set up timed action scripts or escort scripts in SAI, which will take me weeks to learn while not making any improvements to these scripts.

MitchesD commented on sql/updates/world/2016_02_29_29_world.sql:
```diff
@@ -0,0 +1,5 @@
+-- Create gossip_menu_option 7731,7732 for NPC 18588 Floon in Terokkar Forest
+DELETE FROM `gossip_menu_option` WHERE `menu_id` IN (7731,7732) AND `id` = 0;
+INSERT INTO `gossip_menu_option` (`menu_id`,`id`,`option_icon`,`option_text`,`OptionBroadcastTextID`,`option_id`,`npc_option_npcflag`,`action_menu_id`,`action_poi_id`,`box_coded`,`box_money`,`box_text`,`BoxBroadcastTextID`) VALUES
+(7731,0,0,"He wants his "golds." Pay up or die... again...",15564,1,1,0,0,0,0,'',0),
```
this is wrong, you must escape this "golds" otherwise it wont import and travis will still fail

Kittnz replied:
' '

tkrokli replied:
Aha, right. I did not see that before now. Thanks for pointing this out to me.

jackpoz commented:
> By analyzing the blame information on this pull request, we identified @click, @Gacko and @jackpoz to be potential reviewers

can we kill this useless annoying bot ? Otherwise I'll start ignoring every @jackpoz mention I get in the whole TC repo.

Kittnz commented:
It can, https://github.com/facebook/mention-bot but i think the repo owner needs to change this.

Shauren commented:
disabled that bot

Commit:
Core/Scripts: zone_terokkar_forest cleanup

- moved spell and quest IDs into enum
- replaced hardcoded text with DB gossip
- restored correct gossip option order for Skull Pile

tkrokli commented:
Sorry. I lost my work in some overwrite operation when Git Extensions forced me to merge pull instead of pushing my local changes to my own fork. I now have to do it all over again from scratch.
